modules/component-store/src/component-store.ts

```diff
@@ -322,7 +322,7 @@ export class ComponentStore<T extends object> implements OnDestroy {
    * but not used with provideComponentStore()
    */
   private checkProviderForHooks() {
-    asyncScheduler.schedule(() => {
+    asapScheduler.schedule(() => {
       if (
         isDevMode() &&
```

Comment:
Additionally, I'm wondering whether the `isDevMode()` check should be perform at the very beginning, guarding the full `asapScheduler.schedule` call, instead of just scheduled callback body

Reply:
I tried using `isDevMode()` outside the callback body, and it doesn't work as intended.